core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/search/AutocompleteSearchOperationDsl.kt

```diff
@@ -0,0 +1,107 @@
+package com.github.inflab.spring.data.mongodb.core.aggregation.search
+
+import com.github.inflab.spring.data.mongodb.core.annotation.AggregationMarker
+import com.github.inflab.spring.data.mongodb.core.extension.firstOrAll
+import com.github.inflab.spring.data.mongodb.core.extension.toDotPath
+import org.bson.Document
+import kotlin.reflect.KProperty
+
+/**
+ * A Kotlin DSL to build autocomplete search operator using idiomatic Kotlin code.
+ *
+ * @author username1103
+ * @since 1.0
+ * @see <a href="https://www.mongodb.com/docs/atlas/atlas-search/autocomplete">autocomplete</a>
+ */
+@AggregationMarker
+class AutocompleteSearchOperationDsl {
+    private val document = Document()
+
+    /**
+     * The string or strings to search for.
+     * If there are multiple terms in a string, Atlas Search also looks for a match for each term in the string separately.
+     *
+     * @param query The string or strings to search for.
+     */
+    fun query(vararg query: String) {
+        document["query"] = query.toList().firstOrAll()
+    }
+
+    /**
+     * The string or strings to search for.
+     * If there are multiple terms in a string, Atlas Search also looks for a match for each term in the string separately.
+     *
+     * @param query The string or strings to search for.
+     */
+    fun query(query: Iterable<String>) {
+        document["query"] = query.toList().firstOrAll()
+    }
+
+    /**
+     * Indexed autocomplete type of field to search.
+     *
+     * @param path The indexed field or fields to search.
+     * @see <a href="https://www.mongodb.com/docs/atlas/atlas-search/field-types/autocomplete-type/#std-label-bson-data-types-autocomplete">autocomplete</a>
+     */
+    fun path(path: String) {
+        document["path"] = path
+    }
+
+    /**
+     * Indexed autocomplete type of field to search.
+     *
+     * @param path The indexed field or fields to search.
+     * @see <a href="https://www.mongodb.com/docs/atlas/atlas-search/field-types/autocomplete-type/#std-label-bson-data-types-autocomplete">autocomplete</a>
+     */
+    fun path(path: KProperty<String?>) {
+        document["path"] = path.toDotPath()
+    }
+
+    /**
+     * Enable fuzzy search.
+     * Find strings which are similar to the search term or terms.
+     *
+     * @param maxEdits Maximum number of single-character edits required to match the specified search term.
+     * Value can be 1 or 2.
+     * The default value is 2.
+     * Uses [Damerau-Levenshtein distance](https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance).
```

Comment:
I have some question.
How did you find this information?
I could not find it in official docs.

Reply:
Copy from TextSearchOperationDsl fuzzy option.. I'll fix it